arcjet/index.ts

```diff
@@ -945,14 +949,24 @@ export default function arcjet<
     }
 
     const results: ArcjetRuleResult[] = [];
-    // Default all rules to NOT_RUN/ALLOW before doing anything
     for (let idx = 0; idx < rules.length; idx++) {
+      // Default all rules to NOT_RUN/ALLOW before doing anything
       results[idx] = new ArcjetRuleResult({
         ttl: 0,
         state: "NOT_RUN",
         conclusion: "ALLOW",
         reason: new ArcjetReason(),
       });
+
+      // Add top-level characteristics to all Rate Limit rules that don't already have
+      // their own set of characteristics.
+      if (rules[idx].type === "RATE_LIMIT") {
```

Comment:
I believe you can use `isRateLimit()` to [narrow this type](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) to avoid the cast below.

Reply:
Good suggestion, that's much cleaner after I switched it to use the function